### PR TITLE
Document the diff_results script.

### LINKED ISSUE #23: Allow CSV users to produce Krun diffs from warmup_stats wrapper script
Related to #18

### PR DESCRIPTION
**Pull request by snim2**
What it says on the tin.

Fixes #22 
Fixes #23

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,3 +126,26 @@ are:
   * `--wallclock-only` to suppress most details
   * `--xlimits` and `--inset-xlimits` to "zoom in" on parts of the x-axis
 
+
+## Diffing Krun results files
+
+Benchmarking is often performed in order to test whether a change in a given
+VM improves or worsens its performance. Unfortunately, the difference between
+benchmark performance before and after a change is rarely simple. Users will
```
Comment (ltratt) [inline README.md]: The sentence starting "Users" is a bit awkwardly phrased, and I don't think the tables or diffing are restricted to output from Krun. So maybe we should just call this section "Diffing results files" and remove mentions of Krun?

File: README.md [hunk 2]
```diff
@@ -126,3 +126,26 @@ are:
...
+
+The `bin/diff_results` scripts takes two Krun results files as an input and
+produces a LaTeX file (which can be compiled to PDF with pdflatex or similar)
+as output:
+
+```
+./bin/diff_results -r BEFORE.json.bz2 AFTER.json.bz2 -o diff.tex -n 1
+```
+
+The `-n` should normally be set to the number of VMs that were benchmarked
```
Comment (ltratt) [inline README.md]: "normally"?

File: README.md (hunk 1 above)
Comment (snim2) [reply README.md]: At the moment we only diff Krun results files, so anyone with a CSV file needs to convert it first. "Normal" tables can be produced directly from the CSV file with `warmup_stats`.  (see #23)

Comment (snim2) [reply README.md]: Perhaps I should also add some context here, this bit of the README is for people with Krun results files, and the text at the top of the file is for people with CSV files (probably not obvious from the GH diff), so the naming of the section is consistent with the other sections.

Comment (ltratt) [reply README.md]: OK, but that means that we need to think about putting this text elsewhere, or reorganising the file. We want people who might not be using Krun to diff their stats, so we don't want to suggest to them that they can't.

Comment (snim2) [reply README.md]: Right, but that needs some code adding to `warmup_stats`, so it'll come in another PR that fixes #23

Comment (ltratt) [reply README.md]: Oh... Um, maybe we should fix #23 first then. Because I was just about to show this to people who are using CSV files ;)

**Comment by ltratt** [comment]
Where are we with this? I'm keen to see it in, but I think we may be blocked on #23?

**Comment by snim2** [comment]
Yes, I think I pushed a commit to respond to your comments that were actionable here, so we're just waiting on #23

**Comment by snim2** [comment]
This needs to be brought up to date with master to include the latest changes in the `warmup_stats` script. @ltrattt OK to force-push?

**Comment by ltratt** [comment]
Can you not just do a merge commit?

**Comment by snim2** [comment]
OK, let me try

**Comment by snim2** [comment]
This is tested now and ready for re-review

File: README.md [hunk 3]
```diff
@@ -46,6 +46,41 @@ usage:
...
 
+## Creating tables
+
+The `--output-table <file.tex>` flag converts input data into a LaTeX / PDF
+table. Conversion to PDF requires `pdflatex` to be installed. `bin/warmup_stats`
+also needs the names of the language and VM under test, and the output of
+`uname -a` on the machine the benchmarks were run on. Example usage:
+
+```
+./bin/warmup_stats  --output-table table.tex -l javascript -v V8 -u "`uname -a`" results.csv
```
Comment (ltratt) [inline README.md]: Can I suggest we use `bin/warmup_stats` not `./bin/warmup_stats` (and similarly below). `./` might confuse some people.

File: README.md [hunk 4]
```diff
@@ -46,6 +46,41 @@ usage:
...
+
+## Creating diffs
+
+Benchmarking is often performed in order to test whether a change in a given
+VM improves or worsens its performance. Unfortunately, the difference between
+benchmark performance before and after a change is rarely simple. Users will
+want to produce a detailed comparison of the results in Krun results tables
+(above) in order to get a deeper insight into the effects of their changes.
+
+The `--output-diff <file.tex>` flag converts data from exactly two CSV files
```
Comment (ltratt) [inline README.md]: Can we clarify this first sentence? I first read it as "there's one input file called <file.tex>" (and then had to read the rest of the sentence twice to realise I was wrong).

**Comment by snim2** [comment]
Ready for squashing?

**Comment by ltratt** [comment]
Please do ahead!

**Comment by snim2** [comment]
Ah, I should have asked, what happens to the merge commit?

**Comment by ltratt** [comment]
When you rebase, it'll either disappear on its own (excellent!). If it doesn't, you can manually delete it and deal with any merge conflicts that might arise. Cross your fingers it's the former ;)

**Comment by snim2** [comment]
OK, I *think* this makes sense...

**Comment by ltratt** [comment]
Did the merge commit disappear on its own?

**Comment by snim2** [comment]
Yes -- because we had merged another PR this evening I had to `git pull --rebase origin master` anyway.

**Comment by ltratt** [comment]
OK, job's a good 'un.

**Pull request merged**